DataStructures/Graph.java

```diff
@@ -0,0 +1,105 @@
+import java.io.*;
+import java.util.*;
+
+public class Graph {
+    // adjacency list
+    ArrayList<ArrayList<Integer>> adj;
+    int V;
+
+    // constructor
+    public Graph(int V) {
+        this.V = V;
+        adj = new ArrayList<>();
+        for (int i = 0; i < V; i++) {
+            adj.add(new ArrayList<>());
+        }
+    }
+
+    // add edge by addding the vertex to the adjacency list
+    public void addEdge(int u, int v) {
+        adj.get(u).add(v);
+        adj.get(v).add(u);
+    }
+
+    // remove edge by removing the vertex from the adjacency list
+    public void deleteEdge(int u, int v) {
+        adj.get(u).remove((Integer) v);
+        adj.get(v).remove((Integer) u);
+    }
+
+    // breadth first search
+    public void bfs(int s) {
```

Comment:
Please use better variable names. This will increase the readability of your code.

Reply:
Okay, sir.